internal/configs/version1/template_helper.go

```diff
@@ -64,14 +66,131 @@ func makePathWithRegex(path, regexType string) string {
 	}
 }
 
+var setHeader = regexp.MustCompile("^[-A-Za-z0-9]+$")
+
+func validateProxySetHeader(header string) error {
+	header = strings.TrimSpace(header)
+	if !setHeader.MatchString(header) {
+		return errors.New("invalid header syntax - syntax must be header: value")
+	}
+	return nil
+}
+
+func defaultHeaderValues(headerParts []string, headerName string) string {
+	headerValue := strings.TrimSpace(headerParts[0])
+	headerValue = strings.ReplaceAll(headerValue, "-", "_")
+	headerValue = strings.ToLower(headerValue)
+	return fmt.Sprintf("\n\t\tproxy_set_header %s $http_%s;", headerName, headerValue)
+}
+
+func headersGreaterThanOne(headerParts []string, headerName string) string {
+	headerValue := strings.TrimSpace(headerParts[1])
+	return fmt.Sprintf("\n\t\tproxy_set_header %s %q;", headerName, headerValue)
+}
+
+func splitHeaders(header string) ([]string, string) {
+	header = strings.TrimSpace(header)
+	headerParts := strings.SplitN(header, ":", 2)
+	headerName := strings.TrimSpace(headerParts[0])
+	return headerParts, headerName
+}
+
+// minionProxySetHeaders takes a location and a bool map
+// and generates proxy_set_header headers for minions based on the nginx.org/proxy-set-headers annotation in a mergeable ingress.
+// It returns a string containing the generated proxy headers and a map to verify that the header exists
+func minionProxySetHeaders(loc *Location, minionHeaders map[string]bool) (string, map[string]bool, error) {
+	proxySetHeaders, ok := loc.MinionIngress.Annotations["nginx.org/proxy-set-headers"]
+	if !ok {
+		return "", nil, nil
+	}
+	var combinedMinions string
+	headers := strings.Split(proxySetHeaders, ",")
+	for _, header := range headers {
+		headerParts, headerName := splitHeaders(header)
+		err := validateProxySetHeader(headerName)
+		if err != nil {
+			return "", nil, err
+		}
+		if len(headerParts) > 1 {
+			output := headersGreaterThanOne(headerParts, headerName)
+			minionHeaders[headerName] = true
+			combinedMinions += output
+		} else {
+			output := defaultHeaderValues(headerParts, headerName)
+			combinedMinions += output
+		}
+	}
+	return combinedMinions, minionHeaders, nil
+}
+
+// standardIngressOrMasterProxySetHeaders takes two strings, two bools and a bool map
+// and generates proxy_set_header headers based on the nginx.org/proxy-set-headers annotation in a standard ingress.
+// It returns a string containing the generated proxy headers for either standardIngress/NonMergeable or Master.
+func standardIngressOrMasterProxySetHeaders(result string, minionHeaders map[string]bool, proxySetHeaders string, ok bool, isMergeable bool) (string, error) {
+	if !ok {
```

Comment:
Can this `ok` be handled one layer up so it doesnt have to be passed to the `standardIngressOrMasterProxySetHeaders`  function?